### ISSUE TITLE
Unethical IMO

### ISSUE THREAD

**Issue by sankparatkar** [kartikey54/VaccineAvailabilityNotifier#61]
I know not the right place for suggestions. But these automated hits on the Cowin portal could crash the server down and thus are quite selfish. Come on, we all are aware about the servers these guys use.
It is in my opinion a complete missuse of the knowledge since those who actually need the vaccine (45+) won't be able to access the portal. 
So I would say keep this thing private, the problem isn't the code its the numbers who would use it unnecessarily. They just can't keep blocking the IPs or upscaling. This just isn't the time. I hope you understand.

**Comment by hdhruna** [comment]
Maybe adding an option to rate-limit should suffice.

**Comment by sid8840** [comment]
> I know not the right place for suggestions. But these automated hits on the Cowin portal could crash the server down and thus are quite selfish. Come on, we all are aware about the servers these guys use.
> It is in my opinion a complete missuse of the knowledge since those who actually need the vaccine (45+) won't be able to access the portal.
> So I would say keep this thing private, the problem isn't the code its the numbers who would use it unnecessarily. They just can't keep blocking the IPs or upscaling. This just isn't the time. I hope you understand.

Kindly check the scenario 
[COWIN Notifier](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwis0OfzkrDwAhU0muYKHUisBjIQ0PADegQIBhAB&url=https%3A%2F%2Fwww.91mobiles.com%2Fhub%2Fgetjab-covid-vaccine-slot-cowin-notifier%2F&usg=AOvVaw305eH_NmIBHRg5w75Bz8GN)
[Under 45](https://under45.in/)
[Various Other notifier](https://www.moneycontrol.com/news/photos/coronavirus/covid-19-vaccine-appointment-trackers-these-apps-will-notify-you-when-a-vaccination-slot-is-available-nearby-6848561.html)

**Comment by shadabshaikh0** [comment]
I totally agree with @sankparatkar. I think this feature should be integrated into Aarogya Setu App.  Even I created the website with the same idea. But dropped the plan of hosting it after reading @sankparatkar  comment. 👍

**Comment by sankparatkar** [comment]
@shadabshaikh0  appreciate it man. Notifications should always be handled at the server's end. Polling from clients is terrible idea in my view, given we have millions of them. Not sure why they would expose the API to public.smh. 
I think Cowin Dev team should implement this feature which would be quite efficient and easier for all.

**Comment by tejas-2232** [comment]
> I know not the right place for suggestions. But these automated hits on the Cowin portal could crash the server down and thus are quite selfish. Come on, we all are aware about the servers these guys use.
> It is in my opinion a complete missuse of the knowledge since those who actually need the vaccine (45+) won't be able to access the portal.
> So I would say keep this thing private, the problem isn't the code its the numbers who would use it unnecessarily. They just can't keep blocking the IPs or upscaling. This just isn't the time. I hope you understand.

Yes, I agree with you.
We should not block the servers from other real necessary requests.
But in my opinion, if the uses AWS services then the problem might get solved. After all govt infra should be that strong right?
Let me know your view about it.

**Comment by snehitvaddi** [comment]
In my view, both of you are absolutely right. 
* We should be ethical and accountable for doing something in the public domain.
* Govt should also be to the par keeping in correspondence to our country population. (And Govt is doing their best).

**Comment by vedsm** [comment]
The APIs used here are Public APIs(deliberately made so by the CoWIN, so that any 3rd party who can and want to build an expensive notification service around it can do so). Also these APIs have rate limiting plus caching so as to avoid any crashes of CoWIN Portal.

“The [CoWIN site’s vaccine] appointment availability data is cached and may be up to 30 minutes old. Further, these APIs are subject to a rate limit of 100 API calls per 5 minutes per IP.” – [API Setu website](https://apisetu.gov.in/public/marketplace/api/cowin/cowin-public-v2)

**Comment by shadabshaikh0** [comment]
@vedsm Great. Thanks for sharing.

**Comment by sankparatkar** [comment]
@vedsm Yeah i did come across [this](https://apisetu.gov.in/public/marketplace/api/cowin/cowin-public-v2)  couple of days back. But then I checked API used by Production Cowin Portal.  (Image attached)
![Cowin Prod](https://user-images.githubusercontent.com/37218912/117616362-a6615f80-b188-11eb-834a-f074adff865e.PNG)

Then I tried using the same. and I was able to without any issue. 
You can also try at : https://cdn-api.co-vin.in/api/v2/appointment/sessions/calendarByDistrict?district_id=392&date=10-05-2021

1) It does not seem public, it should not be coz it is used in prod. unlike (/v2/appointment/sessions/public/calendarByDistrict)
2) It does not require authentication, it ideally should.
3) This has to be Realtime since it is used in prod.

I don't see why would people not use this instead of public once for notifications.

Anyways, you don't continuously ping Amazon to check what you need is available right, you ask them and they send push notifications. It would be pretty easy to send push notifications by the server once a change is happened rather than millions pinging the server if it has. Polling would give unfair advantage to the once who know about it unless everyone does it (i.e. million requests a second) which is quite terrible in my view. That's why for vaccination purpose, I felt it was unethical.  
 
May be they are not implementing this to avoid blame, that app send us a notification but slots were already full. I don't see any other reason. Let me know if you know. 

Cheers.
